### PR TITLE
Remove hack as integrated builds use FULL_CAFFE2 now

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Review by pjh5 — APPROVED** [review]

**Comment by ezyang** [comment]
Dammit, it doesn't work.

File: .jenkins/caffe2/build.sh [hunk 1]
```diff
@@ -207,22 +194,33 @@ fi
...
+
+  # Configure
+  ${CMAKE_BINARY} "${ROOT_DIR}" ${CMAKE_ARGS[*]} "$@"
+
+  # Build
+  if [ "$(uname)" == "Linux" ]; then
+    make "-j${MAX_JOBS}" install
+  else
+    echo "Don't know how to build on $(uname)"
+    exit 1
```
Comment (pjh5) [inline .jenkins/caffe2/build.sh]: Why not put this as early as possible?

File: .jenkins/caffe2/build.sh [hunk 2]
```diff
@@ -207,22 +194,33 @@ fi
...
-# Build
-if [ "$(uname)" == "Linux" ]; then
-  make "-j${MAX_JOBS}" install
 else
-  echo "Don't know how to build on $(uname)"
-  exit 1
+
+  sudo FULL_CAFFE2=1 python setup.py install
+  # TODO: I'm not sure why this is necessary
+  cp -r torch/lib/tmp_install $INSTALL_PREFIX
```
Comment (pjh5) [inline .jenkins/caffe2/build.sh]: Why is this needed? If setup.py doesn't work by itself then imo it doesn't work.

File: .jenkins/caffe2/build.sh [hunk 3]
```diff
@@ -207,22 +194,33 @@ fi
...
+  fi
 
-# Build
-if [ "$(uname)" == "Linux" ]; then
-  make "-j${MAX_JOBS}" install
 else
-  echo "Don't know how to build on $(uname)"
-  exit 1
+
+  sudo FULL_CAFFE2=1 python setup.py install
```
Comment (pjh5) [inline .jenkins/caffe2/build.sh]: I think you can replace sudo with --user

**Review by pjh5 — APPROVED** [review]

File: .jenkins/caffe2/build.sh (hunk 2 above)
Comment (ezyang) [reply .jenkins/caffe2/build.sh]: Per https://github.com/pytorch/pytorch/pull/10327 this seems to be necessary. I can sign up to fix this later but I need this patch through to unblock tensor merge.
